jaeger_client/sampler.py

```diff
@@ -177,6 +178,100 @@ def __str__(self):
         return 'RateLimitingSampler(%s)' % self.credits_per_second
 
 
+class GuaranteedThroughputProbabilisticSampler(Sampler):
+    """
+    A sampler that leverages both ProbabilisticSampler and RateLimitingSampler.
+    The RateLimitingSampler is used as a guaranteed lower bound sampler such
+    that every operation is sampled at least once in a time interval defined by
+    the lower_bound. ie a lower_bound of 1.0 / (60 * 10) will sample an
+    operation at least once every 10 minutes.
+
+    The ProbabilisticSampler is given higher priority when tags are emitted,
+    ie. if is_sampled() for both samplers return true, the tags for
+    ProbabilisticSampler will be used.
+    """
+    def __init__(self, operation, lower_bound, rate):
+        super(GuaranteedThroughputProbabilisticSampler, self).__init__(
+            tags={
+                SAMPLER_TYPE_TAG_KEY: SAMPLER_TYPE_LOWER_BOUND,
+                SAMPLER_PARAM_TAG_KEY: rate,
+            }
+        )
+        self.probabilistic_sampler = ProbabilisticSampler(rate)
+        self.lower_bound_sampler = RateLimitingSampler(lower_bound)
```

Comment:
please double-check that this class works with values `<1`, as at some point the very first implementation (not sure in which language) only worked with qps expressed as a whole number